client/core/trade.go

```diff
@@ -881,9 +880,9 @@ func (t *trackedTrade) swapMatches(matches []*matchTracker) error {
 	// If these are the last swaps, and the order is filled or canceled or
 	// revoked, then we don't need to lock the change coin.
 	isLastSwaps := len(matches)+alreadySwapped == len(t.matches)
-	isFilled := trade.Filled()+swappedQty == trade.Quantity
-	isCanceled := t.metaData.Status == order.OrderStatusCanceled || t.metaData.Status == order.OrderStatusRevoked
-	skipChange := isLastSwaps && (isFilled || isCanceled)
+	isFilled := trade.Filled() == trade.Quantity
+	notMatchable := t.metaData.Status > order.OrderStatusBooked
+	skipChange := isLastSwaps && (isFilled || notMatchable)
```

Comment:
I would hope that if `isFilled` is true, then `notMatchable` would be true.  Is it possible for `isFilled == true` and `notMatchable == false`?  Seems redundant, but perhaps not.

Reply:
Yep. Just fixed that.